packages/astro/src/content/runtime.ts

```diff
@@ -618,15 +622,22 @@ export function createReference({ lookupMap }: { lookupMap: ContentLookupMap })
 				}),
 			])
 			.transform(
-				async (
+				(
 					lookup:
 						| string
 						| { id: string; collection: string }
 						| { slug: string; collection: string },
 					ctx,
 				) => {
+					if (!store) {
+						ctx.addIssue({
+							code: ZodIssueCode.custom,
+							message: `**${ctx.path.join('.')}:** Reference to ${collection} could not be resolved. Store not available.`,
```

Comment:
I would add a more actionable message. If the user encounters this error, what should they do? File an issue? Purge the store? We should provide an escape hatch to our users